### PR TITLE
Updated mesh quality metrics for TMOP-based mesh optimization

### PR DESCRIPTION
**Pull request by kmittal2**
This PR updates the mesh quality metrics for TMOP-based mesh optimization.

The snapshot below shows a comparison for discrete adaptivity using shape+size metrics `mu_7=|T-T^-t|^2` (left) and newly added polyconvex metrix `mu_80=(1-gamma)mu_2 + gamma mu_77` (right) with `gamma=0.9`. Here `mu_2 = 0.5|T|^2/tau-1` and `mu_77=0.5(tau-1/tau)^2`. In this example, the TMOP objective function is reduced by about 36% with `mu_7` and 68% with `mu_80`.
image

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1861](https://github.com/mfem/mfem/pull/1861) | @kmittal2 | @tzanio | @vladotomov + @jandrej | 11/03/20 | 11/18/20 | 11/22/20 | |

### CODE REVIEW

**Review by jandrej — COMMENTED** [review]
LGTM

File: fem/tmop.hpp [hunk 1]
```diff
@@ -316,7 +345,34 @@ class TMOP_Metric_077 : public TMOP_QualityMetric
...
+   TMOP_Metric_080(double gamma_) : gamma(gamma_),
+      sh_metric(new TMOP_Metric_002),
+      sz_metric(new TMOP_Metric_077)
+   {
+      // (1-gamma) mu_2 + gamma mu_77
+      AddQualityMetric(sh_metric, 1.-gamma_);
+      AddQualityMetric(sz_metric, gamma_);
+   }
+
+   virtual void SetTargetJacobian(const DenseMatrix &_Jtr)
```
Comment (vladotomov) [inline fem/tmop.hpp:366]: Shouldn't be needed.

**Comment by vladotomov** [comment]
@kmittal2 could you paste the preconditioner options for MINRES in this PR? See `mesh-optimizer` and `pmesh-optimizer` in the branch `tmopt`.

**Comment by jandrej** [comment]
@adrienbernede I tried to trigger gitlab ci here, but nothing happened for a full day now.

**Comment by adrienbernede** [comment]
LGTM

**Review by jandrej — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Review by tzanio — APPROVED** [review]
Thanks for this update @k10!

@vladotomov and @jandrej -- thank you for reviewing!

**Pull request merged**